.github/workflows/upload-and-deploy.yaml

```diff
@@ -25,6 +25,59 @@ jobs:
       - uses: actions/setup-python@v4
       - uses: pre-commit/action@v3.0.0
 
+  deploy-snowflake:
+    name: Deploy Snowflake resources
+    needs: pre-commit
+    runs-on: ubuntu-latest
+    env:
+      PRIVATE_KEY_PASSPHRASE: ${{ secrets.SNOWFLAKE_PRIVATE_KEY_PASSPHRASE }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Configure Snowflake connection
+        run: |
+          # Create temporary files for config.toml and our private key
+          #mkdir ~/.snowflake
+          #config_file="~/.snowflake/config.toml"
+          config_file=$(mktemp)
+          private_key_file=$(mktemp)
+
+          # Write to the private key file
+          echo "${{ secrets.SNOWFLAKE_PRIVATE_KEY }}" > $private_key_file
```

Comment:
Does `SNOWFLAKE_PRIVATE_KEY` and `SNOWFLAKE_PRIVATE_KEY_PASSPHRASE` ever expire?

Reply:
They don't expire. Snowflake [supports rotating the key pairs](https://docs.snowflake.com/en/user-guide/key-pair-auth#configuring-key-pair-rotation), but I don't know if we have this configured.